sensor/common/image/service_impl_test.go

```diff
@@ -0,0 +1,280 @@
+			expectCache:      expectCacheHelper(suite.mockCache, 1, createScannedImage(imageName, imageID)),
+			expectedError:    nil,
+			expectedResponse: createImageResponse(imageName, imageID),
+		},
+		"Cache miss and central is unreachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventOfflineMode,
+			expectCache:      expectCacheHelper(suite.mockCache, 1, nil),
+			expectedError:    errCentralNoReachable,
+			expectedResponse: nil,
+		},
+		"Cache miss and central is reachable": {
+			request:           createImageRequest(imageName, imageID, false),
+			notify:            common.SensorComponentEventCentralReachable,
+			expectCache:       expectCacheHelper(suite.mockCache, 1, nil),
+			expectRegistry:    expectRegistryHelper(suite.mockRegistryStore, 1, false),
+			expectCentralCall: expectCentralCall(suite.mockCentral, 1, createScanImageInternalResponse(imageName, imageID), nil),
+			expectedError:     nil,
+			expectedResponse:  createImageResponse(imageName, imageID),
+		},
+		"Cache miss, central is reachable and returns error": {
+			request:           createImageRequest(imageName, imageID, false),
+			notify:            common.SensorComponentEventCentralReachable,
+			expectCache:       expectCacheHelper(suite.mockCache, 1, nil),
+			expectRegistry:    expectRegistryHelper(suite.mockRegistryStore, 1, false),
+			expectCentralCall: expectCentralCall(suite.mockCentral, 1, nil, err),
+			expectedError:     errCentral,
+			expectedResponse:  nil,
+		},
+		"Cache miss, local scan, central is unreachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventOfflineMode,
+			expectCache:      expectCacheHelper(suite.mockCache, 1, nil),
+			expectedError:    errCentralNoReachable,
+			expectedResponse: nil,
+		},
+		"Cache miss, local scan, central is reachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventCentralReachable,
+			expectCache:      expectCacheHelper(suite.mockCache, 1, nil),
+			expectRegistry:   expectRegistryHelper(suite.mockRegistryStore, 1, true),
+			expectLocalScan:  expectLocalScan(suite.mockLocalScan, 1, createScannedImage(imageName, imageID), nil),
+			expectedError:    nil,
+			expectedResponse: createImageResponse(imageName, imageID),
+		},
+		"Cache miss, local scan returns error, central is reachable": {
+			request:          createImageRequest(imageName, imageID, false),
+			notify:           common.SensorComponentEventCentralReachable,
+			expectCache:      expectCacheHelper(suite.mockCache, 1, nil),
+			expectRegistry:   expectRegistryHelper(suite.mockRegistryStore, 1, true),
+			expectLocalScan:  expectLocalScan(suite.mockLocalScan, 1, nil, err),
+			expectedError:    errLocalScan,
+			expectedResponse: nil,
+		},
+	}
+	for testName, c := range cases {
+		suite.T().Run(testName, func(t *testing.T) {
```

Comment:
```suggestion
		suite.Run(testName, func() {
```
You can call `Run` directly from `suite` and drop the `t` parameter.

Reply:
Done